pkg/ui/workspaces/db-console/src/views/reports/containers/customChart/index.tsx

```diff
@@ -208,56 +208,79 @@ export class CustomChart extends React.Component<
     );
   };
 
+  getSources = (
+    nodesSummary: NodesSummary,
+    metricState: CustomMetricState,
+  ): string[] => {
+    if (!(nodesSummary?.nodeStatuses?.length > 0)) {
+      return [];
+    }
+    // If we have no nodeSource, and we're not asking for perSource metrics,
+    // then the user is asking for cluster-wide metrics. We can return an empty
+    // source list.
+    if (metricState.nodeSource === "" && !metricState.perSource) {
+      return [];
+    }
+    if (isStoreMetric(nodesSummary.nodeStatuses[0], metricState.metric)) {
+      // If a specific node is selected, return the storeIDs associated with that node.
+      // Otherwise, we're at the cluster level, so we grab each store ID.
+      return metricState.nodeSource
+        ? nodesSummary.storeIDsByNodeID[metricState.nodeSource]
+        : Object.values(nodesSummary.storeIDsByNodeID).flatMap(s => s);
```

Comment:
nit: Is `Object.values(nodesSummary.storeIDsByNodeID).flatMap(s => s);` the equivalent of `Object.values(nodesSummary.storeIDsByNodeID).flat();`

For context, is `Object.values()` returning arrays?

Reply:
It is, and in retrospect I should have just used `.flat()` here 😅.

`Object.values()` in this case is returning a 2D array, where each array represents the store IDs for a given node. By flattening the map we get the complete list of store IDs. To be honest, I'm a JS newbie and I didn't realize `.flat()` existed. `.flatMap()` is just using the identity function, so it wasn't really necessary. TIL!